### PR TITLE
Fix error from get_client() with no global client

### PR DESCRIPTION
**Pull request by li-dan**
Fix TypeError from get_client() with no global client. This is supposed
to raise a ValueError.

### CODE REVIEW

File: distributed/tests/test_client.py [hunk 1]
```diff
@@ -4612,6 +4612,12 @@ def f(x):
         del distributed.tmp_client
 
 
+def test_get_client_no_cluster():
+    msg = u'No global client found and no address provided'
```
Comment (mrocklin) [inline distributed/tests/test_client.py]: Seems fine to me.  I recommend dropping the unicode string prefix here.  The dask codebase defaults to Python 3.

Comment (li-dan) [reply distributed/tests/test_client.py]: Okay, fixed.

**Comment by mrocklin** [comment]
+1 from me.  Waiting for travis-ci

**Comment by li-dan** [comment]
Some of the builds failed. It seems that `get_worker` is not raising `ValueError`. Is there a way to set up the test so that there are no global workers?

**Comment by mrocklin** [comment]
> Some of the builds failed. It seems that get_worker is not raising ValueError. Is there a way to set up the test so that there are no global workers?

Hrm, that's unfortunate.  Two possible solutions come to mind

```python
def test_foo():
    from distributed.workers import _global_workers
    _global_workers.clear()
    # the rest of your code
```

Or, if we wanted to do this more holistically, we would modify the `gen_cluster` and `cluster` functions in `utils_test.py` to always clear this collection after every test.  *Almost* all tests operate with one of these functions, so this is a nice way to enforce hygiene across the test suite.

**Comment by li-dan** [comment]
I've made the smaller change for now. Let's see if that fixes the builds.

**Comment by mrocklin** [comment]
It looks like this fails on Python 2.  My apologies for the bad suggestion.  You probably want the following instead:

    del _global_workers[:]

**Comment by li-dan** [comment]
I made the fix, and all builds are passing now.

**Comment by mrocklin** [comment]
Awesome.  Thank you for following through with this @li-dan !  Merging.

**Pull request merged**